src/main/java/uk/gov/hmcts/reform/civil/handler/tasks/HearingFeePaidHandler.java

```diff
@@ -0,0 +1,55 @@
+package uk.gov.hmcts.reform.civil.handler.tasks;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.camunda.bpm.client.task.ExternalTask;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.civil.enums.PaymentStatus;
+import uk.gov.hmcts.reform.civil.event.HearingFeeUnpaidEvent;
+import uk.gov.hmcts.reform.civil.event.StrikeOutEvent;
+import uk.gov.hmcts.reform.civil.helpers.CaseDetailsConverter;
+import uk.gov.hmcts.reform.civil.model.CaseData;
+import uk.gov.hmcts.reform.civil.service.search.CaseHearingFeePaidSearchService;
+
+import java.util.List;
+
+@Slf4j
+@RequiredArgsConstructor
+@Component
+public class HearingFeePaidHandler implements BaseExternalTaskHandler {
+
+    private final CaseHearingFeePaidSearchService caseSearchService;
+    private final ApplicationEventPublisher applicationEventPublisher;
+    private final CaseDetailsConverter caseDetailsConverter;
+
+    @Override
+    public void handleTask(ExternalTask externalTask) {
+        List<CaseDetails> cases = caseSearchService.getCases();
+        log.info("Job '{}' found {} case(s)", externalTask.getTopicName(), cases.size());
+
+        cases.forEach(caseDetails -> {
+            try {
+                CaseData caseData = caseDetailsConverter.toCaseData(caseDetails);
+
+                if (caseData.getHearingDueDate() == null
+                    || caseData.getHearingFeePaymentDetails().getStatus() == PaymentStatus.SUCCESS) {
+                    log.info("Current case status '{}'", caseDetails.getState());
```

Comment:
I'm a bit confused, Suvarna - where is the logic that takes us along the positive path?
The 'Strike Out' event was meant for the negative path, though I replaced that with the more descriptive 'HearingFeeUnpaid' event.